pontoon/base/models.py

```diff
@@ -3203,6 +3203,30 @@ def map_entities(
         return entities_array
 
 
+class ChangedEntityLocaleManager(models.Manager):
+    def bulk_mark_changed(self, translations, locale=None):
```

Comment:
Sorry, I mislead you with my note in https://github.com/mozilla/pontoon/issues/2771#issue-1603963121. Since Translation QuerySet is used as a parameter, it would make more sense to move this logic under `TranslationQuerySet`. 

The API call would then be simply `translations.bulk_mark_changed()`, without any parameters, because we can get locale from each translation object. Before generating the translations QuerySet, we'd need to prefetch related `entity__resource__project` data (and revert prefetching established in #2765).

Going forward, we should file an issue to also move `mark_changed()` [model method](https://docs.djangoproject.com/en/4.1/topics/db/models/#model-methods.) from Entity to Translation model. It's currently always used as `translation.entity.mark_changed()`. And finally, migrate ChangedEntityLocale model to ChangedTranslation. The reason this is currently bound to Entity rather than Translation model is a relic of the past when we had the ability to delete translations without rejecting them first.